### ISSUE TITLE
How is are Background Jobs implemented with this library?

### ISSUE THREAD

**Issue by justinjalandoniiwest** [fluentscheduler/FluentScheduler#297]
I would like to know an overview of how background jobs run? Is a new Thread just running?

**Comment by ghost** [comment]
> Is a new Thread just running?

Yes, that's correct. But there's more to it, it's not running inside a single long live thread. The scheduler basically creates a new thread to run your job every time according to your schedule. So you better read about [Reentrancy](https://fluentscheduler.github.io/reentrancy/) to prevent **parallel with a previously triggered execution of the same schedule**. This library requires knowledge in multi-threading, if you know this subject you'll find this library very powerful. I've used this library in so many real-world projects that I delivered to customers.

**Comment by AraHaan** [comment]
@dahanj95 does that include ASP.NET MVC projects using OWIN?
